### PR TITLE
Add keymap completion

### PR DESCRIPTION
**Pull request by dkearns**
Problem:  Keymap completion is not available.
Solution: Add keymap completion.

Add keymap completion to the 'keymap' option, user commands and builtin
completion functions.

### CODE REVIEW

**Review by yegappan — APPROVED** [review]

File: src/Makefile [hunk 1]
```diff
@@ -955,6 +955,7 @@ TOOLS = xxd/xxd$(EXEEXT)
 #
 # Uncomment the next line to install Vim in your home directory.
 #prefix = $(HOME)
+prefix = $(HOME)/opt
```
Comment (benknoble) [inline src/Makefile]: This probably isn't supposed to make it into the final version.

File: runtime/doc/builtin.txt [hunk 2]
```diff
@@ -3566,6 +3566,7 @@ getcompletion({pat}, {type} [, {filtered}])		*getcompletion()*
 		help		help subjects
 		highlight	highlight groups
 		history		|:history| suboptions
+		keymap		keyboard mappings
```
Comment (benknoble) [inline runtime/doc/builtin.txt:3569]: A note that this is only available with `|+keymap|` or similar might be good.

File: runtime/doc/map.txt [hunk 3]
```diff
@@ -1629,6 +1629,7 @@ completion can be enabled:
 	-complete=help		help subjects
 	-complete=highlight	highlight groups
 	-complete=history	:history suboptions
+	-complete=keymap	keyboard mappings
```
Comment (benknoble) [inline runtime/doc/map.txt:1632]: Ditto: a note that this is only available with `|+keymap|` or similar might be good.

File: src/Makefile (hunk 1 above)
Comment (benknoble) [reply src/Makefile]: Hm. Something is fishy: I don't seem able to mark this comment resolved. But it is. :)

File: runtime/doc/builtin.txt (hunk 2 above)
Comment (dkearns) [reply runtime/doc/builtin.txt:3569]: Thanks.  As it's not done for the other completion types in these lists I took the opportunity to deliberately not do so for this addition.

I'll look at updating them in a followup PR if I can discern a pattern in the help.

**Comment by chrisbra** [comment]
okay thanks. However I'll move this to after the 9.1 release

**Pull request closed without merging**